packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -144,12 +129,14 @@ export class JaegerExporter implements SpanExporter {
     // Flush all spans on each export. No-op if span buffer is empty
     await this._flush();
 
-    if (done) return done({ code: ExportResultCode.SUCCESS });
+    if (done) return process.nextTick(done, { code: ExportResultCode.SUCCESS });
   }
 
   private async _append(span: jaegerTypes.ThriftSpan): Promise<number> {
+    const sender = this._getSender(span);
+
     return new Promise((resolve, reject) => {
-      this._sender.append(span, (count: number, err?: string) => {
+      sender.append(span, (count: number, err?: string) => {
```

Comment:
```suggestion
      this._getSender(span).append(span, (count: number, err?: string) => {
```

and remove line 136.
This way you will not need the local reference to the sender